kratos/sources/model_part.cpp

```diff
@@ -1616,28 +1632,71 @@ void ModelPart::RemoveGeometryFromAllLevels(std::string GeometryName)
 ///@name Sub Model Parts
 ///@{
 
-ModelPart&  ModelPart::CreateSubModelPart(std::string const& NewSubModelPartName)
+ModelPart& ModelPart::CreateSubModelPart(std::string const& NewSubModelPartName)
 {
-    // Here a warning would be enough. To be disscussed. Pooyan.
-    KRATOS_ERROR_IF(mSubModelParts.find(NewSubModelPartName) != mSubModelParts.end())
-        << "There is an already existing sub model part with name \"" << NewSubModelPartName
-        << "\" in model part: \"" << Name() << "\"" << std::endl;
+    const auto sub_model_parts_list = SplitSubModelPartHierarchy(NewSubModelPartName);
+    if (sub_model_parts_list.size() == 1) {
+        // Here a warning would be enough. To be disscussed. Pooyan.
+        KRATOS_ERROR_IF(mSubModelParts.find(NewSubModelPartName) != mSubModelParts.end())
+            << "There is an already existing sub model part with name \"" << NewSubModelPartName
+            << "\" in model part: \"" << Name() << "\"" << std::endl;
+
+        ModelPart* praw = new ModelPart(NewSubModelPartName, this->mpVariablesList, this->GetModel());
+        Kratos::shared_ptr<ModelPart> p_model_part(praw); //we need to construct first a raw pointer
+        p_model_part->SetParentModelPart(this);
+        p_model_part->mBufferSize = this->mBufferSize;
+        p_model_part->mpProcessInfo = this->mpProcessInfo;
+        mSubModelParts.insert(p_model_part);
+        return *p_model_part;
 
-    ModelPart* praw = new ModelPart(NewSubModelPartName, this->mpVariablesList, this->GetModel());
-    Kratos::shared_ptr<ModelPart>  p_model_part(praw); //we need to construct first a raw pointer
-    p_model_part->SetParentModelPart(this);
-    p_model_part->mBufferSize = this->mBufferSize;
-    p_model_part->mpProcessInfo = this->mpProcessInfo;
-    mSubModelParts.insert(p_model_part);
-    return *p_model_part;
+    } else {
+        ModelPart* p_current_mp = this;
+        for (const auto& r_smp_name : sub_model_parts_list) {
+            if (!p_current_mp->HasSubModelPart(r_smp_name)) {
+                p_current_mp = &p_current_mp->CreateSubModelPart(r_smp_name);
+            } else {
+                p_current_mp = &p_current_mp->GetSubModelPart(r_smp_name);
+            }
+        }
+        return *p_current_mp;
```

Comment:
One problem which I see in here is, if someone requests to create a submodel part with `NewSubModelPartName = "model_part_1.model_part_2"` where both alreaady exists, then this algorithm will return the existing `model_part_2`, and not throwing an error which is inconsistent with the call `model_part_1.CreateSubModelPart("model_part_2")`. Is there a specific reason why you don't do it in a recursive manner? Then this won't be a problem. Or may be to loop until `N-1` and always use `CreateSubModelPart` on the last.